### ISSUE TITLE
FrameBuffer multisampling support

### ISSUE THREAD

**Issue by IgorMats** [cocos2d/cocos2d-x#16820]
Can someone please add multisampling support for **experimental::FrameBuffer** and render target to texture!? I think it's really awesome feature for post processing.
I have tried to do it by myself but always get opengl error 0x0506.

Can please someone help me?

**Comment by dumganhar** [comment]
Yes, we need this feature.
Currently, we even don't have a test case for FrameBuffer class although it's in experimental namespace.

Could you share your work about `multisampling`? I will be glad to check it out why it causes the `opengl error 0x0506`.

**Comment by IgorMats** [comment]
@dumganhar Do you know is it true that GL_TEXTURE_2D can't be multisampled like described [here](http://stackoverflow.com/questions/6342126/opengl-render-to-texture-in-iphone-fails-when-msaa-is-enabled/6343055#6343055) so I can use multisampling only with RenderTargetRenderBuffer not RenderTarget?

**Comment by IgorMats** [comment]
It seems to be true so I have achieved this by create 2 frame buffers, one with render buffer target and another one with texture render target and then transfer render buffer to the texture of second frame buffer by calling **glResolveMultisampleFramebufferAPPLE** so I can't make pull request because this solution is not cross platform solution.

Maybe somebody can help me with Android?

Thanks.

**Comment by dumganhar** [comment]
Yes, I figured that `GL_TEXTURE_2D_MULTISAMPLE` isn't supported in OpenGLES, RenderTarget (which means rendering to texture target in cocos2d-x) probably will not work with multisampled feature.

> It seems to be true so I have achieved this by create 2 frame buffers, one with render buffer target and another one with texture render target and then transfer render buffer to the texture of second frame buffer by calling glResolveMultisampleFramebufferAPPLE so I can't make pull request because this solution is not cross platform solution.

Good to know it works on iOS/Mac now. Yep, glResolveMultisampleFramebufferAPPLE is an apple specific gl extension function, we can't use it for other platforms.
Could you upload test code which works on iOS/Mac to your github branch?
I guess someone will help you to port it to other platforms.

**Comment by IgorMats** [comment]
@dumganhar I will try to support another platforms by myself and then will make pull request.
If anyone can or would like to help me please write some message so we will can collaborate :) Thanks!
